src/org/opensha/commons/calc/magScalingRelations/magScalingRelImpl/TMG2017SubMagAreaRel.java

```diff
@@ -0,0 +1,180 @@
+/*******************************************************************************
+ * Copyright 2009 OpenSHA.org in partnership with
+ * the Southern California Earthquake Center (SCEC, http://www.scec.org)
+ * at the University of Southern California and the UnitedStates Geological
+ * Survey (USGS; http://www.usgs.gov)
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package org.opensha.commons.calc.magScalingRelations.magScalingRelImpl;
+
+import org.opensha.commons.calc.magScalingRelations.MagAreaRelationship;
+import org.opensha.commons.exceptions.InvalidRangeException;
+import org.opensha.commons.util.FaultUtils;
+
+/**
+ * <b>Title:</b>TMG2017_MagAreaRel<br>
+ *
+ * <b>Description:</b>
+ * <p>
+ * <b>Description:</b>
+ * <p>
+ * This implements SUBDUCTION specific magnitude versus rupture area relations
+ * of Thingbaijam K.K.S., P.M. Mai and K. Goda 2017, Bull. Seism. Soc. Am., 107,
+ * 2225–2246.
+ * 
+ * 
+ * We consider rake to differentiate different broad faulting-types: interface
+ * (reverse-faulting) and inslab (normal-faulting). The classification is as
+ * follows: Interface (reverse-faulting): Rake angles within 45 to 135, and
+ * Inslab (normal-faulting): Rake angles within -45 to -135.
+ * 
+ * Notes: [1] Relations for strike-slip events are not defined. [2] The standard
+ * deviation for area as a function of mag is given for log(area) (base-10) not
+ * area.
+ * 
+ * Also see: https://github.com/thingbaijam/sceqsrc
+ * </p>
+ *
+ * @version 0.0
+ */
+
+public class TMG2017SubMagAreaRel extends MagAreaRelationship {
+
+	final static String C = "TMG2017SubMagAreaRel";
+	public final static String NAME = "Thingbaijam et al.(2017)";
+
+	/**
+	 * Computes the median magnitude from rupture area for previously set rake
+	 * values that distinguish between interface (reverse) events and inslab normal
+	 * faulting events .
+	 * 
+	 * @param area in km
+	 * @return median magnitude MW
+	 */
+	public double getMedianMag(double area) {
+
+		// FaultUtils.assertValidRake(rake) does not invalidate NaN - :(
+		if (Double.isNaN(rake)) {
+			// return NaN because this estimate is not available.
+			return Double.NaN;
+		}
+
+		if ((rake <= 45 && rake >= -45) || rake >= 135 || rake <= -135) {
```

Comment:
This behaviour is different to the one of the combined implementation. Is this one the correct one?

Reply:
Yes. here, we do not have "strike-slip" for subduction events.